clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -633,6 +641,9 @@ private void handleListOffsetResponse(TopicPartition topicPartition,
             completedFetch.metricAggregator.record(tp, bytes, recordsCount);
         }
 
+        if (!emptyResponse)
+            subscriptions.movePartitionToEnd(tp);
```

Comment:
Meta-comment: what would really be the impact if the consumer didn't do any reordering? Suppose there was a partition which was effectively being starved by always being ordered last. The only way it would stay starved indefinitely is if the message load from the partitions ordered before it was so high that it took all the consumer's effort just to keep up with them. In effect, this is a situation where the load from all partitions is simply too much for the consumer and some lag is inevitable. Maybe we could argue that it's better to reorder partitions fairly in this case so that the lag is spread more evenly? But really this is an exceptional case and if it persists long enough (that is, if the load doesn't drop), it means the user probably needs to add more consumers to the group. Given that, I'm wondering if doing a random shuffling is "good enough" in practice to ensure that lag is distributed when the consumer is not keeping up? We could even shuffle among the topics first and then among the partitions for each topic to keep the request size low. What do you think?